docs/advanced/extensibleenums.md

```diff
@@ -7,216 +7,139 @@ Extensible Enums are an enhancement of specific Vanilla enums to allow new entri
-        throw new IllegalStateException("Enum not extended");
-    }
-
-    // ...
+  ]
 }
 ```
 
-### The `init` method
-
-Sometimes, an enum will do something with its constants or values after registration. However, as these instances statically call the associated entries, the values added via `create` may not be referenced. To get around this, extensible enums can override the `init` method to do any post-constructor setup required by the specific enum constant.
-
 ```java
-// As an example
-public enum ExampleEnumInit {
-    VALUE_1(0, "value_1", false),
-    VALUE_2(1, "value_2", true);
-
-    private static final Map<String, boolean> ARG2_TO_ARG3 =
-        Arrays.stream(ExampleEnumInit.values())
-        .collect(
-            Collectors.toMap(
-                e -> e.arg2,
-                e -> e.arg3
-            )
-        );
-
-    ExampleEnumInit(int arg1, String arg2, boolean arg3) {
-        // ...
-    }
-}
-
-// The final extended enum may look something like this
-public enum ExampleEnumInit implements net.neoforged.neoforge.common.IExtensibleEnum {
-    VALUE_1(0, "value_1", false),
-    VALUE_2(1, "value_2", true);
-
-    private static final Map<String, boolean> ARG2_TO_ARG3 =
-        Arrays.stream(ExampleEnumInit.values())
-        .collect(
-            Collectors.toMap(
-                e -> e.arg2,
-                e -> e.arg3
-            )
-        );
-
-    ExampleEnumInit(int arg1, String arg2, boolean arg3) {
-        // ...
-    }
-
-    // Matches the constructor
-    public static ExampleEnumInit create(String name, int arg1, String arg2, boolean arg3) {
-        throw new IllegalStateException("Enum not extended");
-    }
-
-    @Override
-    public void init() {
-        // Gets called for every new enum
-        ARG2_TO_ARG3.put(this.arg2, this.arg3);
+public class MyEnumParams {
+    public static final EnumProxy<Rarity> CUSTOM_RARITY_ENUM_PROXY = new EnumProxy<>(
+            Rarity.class, -1, "examplemod:custom", (UnaryOperator<Style>) style -> style.withItalic(true)
+    );
+    
+    public static Object getTestDamageEffectsParameter(int idx, Class<?> type) {
+        return type.cast(switch (idx) {
+            case 0 -> "examplemod:test";
+            case 1 -> (Supplier<SoundEvent>) () -> SoundEvents.DONKEY_ANGRY;
+            default -> throw new IllegalArgumentException("Unexpected parameter index: " + idx);
+        });
     }
 }
 ```
 
-### Codecs
+#### Constructor
 
-[Enum codecs][codec] generally take in the `values` array and resolve the codec instantly. As the array is evaluated before any mod entries are added, no mod entries will be supported. There are two solutions to this problem, which to choose depends on whether the enum implements the `StringRepresentable` interface or not.
+The constructor must be specified as a method descriptor and must only contain the parameters visible in the source code, omitting the hidden constant name and ordinal parameters.
```

Comment:
Probably a link to the JVMS page similar to Access transformers on how to write out the definition.